### PR TITLE
isDate function to TypeScript

### PR DESCRIPTION
**Pull request by tan75**
#### Changes
Migrate isDate() to TS
Fix bug - see discussion #2338

This closes https://github.com/date-fns/date-fns-typescriptify/issues/66

### CODE REVIEW

**Review by kossnocorp — CHANGES_REQUESTED** [review]
Please remove the fix for #2338 from the PR. See: https://github.com/date-fns/date-fns/discussions/2338#discussioncomment-550803

**Comment by tan75** [comment]
> Please remove the fix for #2338 from the PR. See: [#2338 (comment)](https://github.com/date-fns/date-fns/discussions/2338#discussioncomment-550803)

Done.

File: src/isDate/index.ts [hunk 1]
```diff
@@ -12,31 +12,31 @@ import requiredArgs from '../_lib/requiredArgs/index'
  *
  * - [Changes that are common for the whole library](https://github.com/date-fns/date-fns/blob/master/docs/upgradeGuide.md#Common-Changes).
  *
- * @param {*} value - the value to check
+ * @param {any} value - the value to check
```
Comment (kossnocorp) [inline src/isDate/index.ts:15]:
```suggestion
 * @param {unknown} value - the value to check
```

File: src/isDate/index.ts [hunk 2]
```diff
@@ -12,31 +12,31 @@ import requiredArgs from '../_lib/requiredArgs/index'
...
  * //=> false
  *
  * @example
  * // For an object:
- * var result = isDate({})
+ * const result = isDate({})
  * //=> false
  */
-export default function isDate(value) {
+export default function isDate(value: any): value is Date {
```
Comment (kossnocorp) [inline src/isDate/index.ts:39]:
```suggestion
export default function isDate(value: unknown): value is Date {
```

**Review by kossnocorp — APPROVED** [review]
🙏

**Pull request merged**